projects/extension/src/types/index.ts

```diff
@@ -17,18 +17,35 @@ export type uApp = {
     enabled: boolean;
 }
 
-interface ChainSpec {
+export interface Client {
   name: string;
   icon?: string;
   status: NetworkStatus;
 }
-export interface Network extends ChainSpec {
+export interface Network extends Client {
   chain: smoldot.SmoldotChain;
   tabId: number;
   parachains?: Parachain[];
 }
```

Comment:
At a first glance, I would say that it's probably best to use composition rather than having the `Network` interface extend `Client`. I.e I think this would be better:

```ts
export interface Network {
  client: Client,
  chain: smoldot.SmoldotChain;
  tabId: number;
  paracahins?: Parachain[];
}
```

Also, wouldn't it make more sense for the `parachains` prop to not be optional and use an empty array when there are no `parachains`? IMO that usually makes the API more ergonomic to work with.

Reply:
As discussed - I understand the confusion. I already altered the words to make more sense